cmd/operator-sdk/add/api.go

```diff
@@ -21,89 +21,268 @@ import (
+		&ansible.RolesDefaultsMain{Resource: *r},
+		&ansible.RolesTasksMain{Resource: *r},
+		&ansible.RolesHandlersMain{Resource: *r},
+	)
+	if err != nil {
+		log.Fatalf("API scaffold failed: %v", err)
+	}
+	if err = genutil.GenerateCRDNonGo("", *r, apiFlags.CrdVersion); err != nil {
+		return err
+	}
+	// update deploy/role.yaml for the given resource r.
+	if err := scaffold.UpdateRoleForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the RBAC manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+	return nil
+
+}
+
+// TODO
+// Consolidate scaffold func to be used by both "new" and "add api" commands.
+func doHelmAPIScaffold() error {
+	// Create and validate new resource.
+	/*r, err := scaffold.NewResource(flagSet.APIVersion, flagSet.Kind)
+	if err != nil {
+		return err
+	}*/
+	absProjectPath := projutil.MustGetwd()
+	projectName := filepath.Base(absProjectPath)
+	cfg := &input.Config{
+		AbsProjectPath: absProjectPath,
+		ProjectName:    projectName,
+	}
+
+	createOpts := helm.CreateChartOptions{
+		ResourceAPIVersion: apiFlags.APIVersion,
+		ResourceKind:       apiFlags.Kind,
+		Chart:              apiFlags.HelmChartRef,
+		Version:            apiFlags.HelmChartVersion,
+		Repo:               apiFlags.HelmChartRepo,
+	}
+
+	r, chart, err := helm.CreateChart(cfg.AbsProjectPath, createOpts)
+	if err != nil {
+		return fmt.Errorf("failed to create helm chart: %v", err)
+	}
+
+	valuesPath := filepath.Join("<project_dir>", helm.HelmChartsDir, chart.Name(), "values.yaml")
+
+	rawValues, err := yaml.Marshal(chart.Values)
+	if err != nil {
+		return fmt.Errorf("failed to get raw chart values: %v", err)
+	}
+	crSpec := fmt.Sprintf("# Default values copied from %s\n\n%s", valuesPath, rawValues)
+
+	// update watch.yaml for the given resource r.
+	if err := helm.UpdateHelmWatchForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the Watch manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+
+	s := &scaffold.Scaffold{}
+	err = s.Execute(cfg,
+		&scaffold.CR{
+			Resource: r,
+			Spec:     crSpec,
+		},
+	)
+	if err != nil {
+		log.Fatalf("API scaffold failed: %v", err)
+	}
+	if err = genutil.GenerateCRDNonGo("", *r, apiFlags.CrdVersion); err != nil {
 		return err
 	}
 
-	log.Infof("Generating api version %s for kind %s.", apiVersion, kind)
+	roleScaffold := helm.DefaultRoleScaffold
+	if k8sCfg, err := config.GetConfig(); err != nil {
+		log.Warnf("Using default RBAC rules: failed to get Kubernetes config: %s", err)
+	} else if dc, err := discovery.NewDiscoveryClientForConfig(k8sCfg); err != nil {
+		log.Warnf("Using default RBAC rules: failed to create Kubernetes discovery client: %s", err)
+	} else {
+		roleScaffold = helm.GenerateRoleScaffold(dc, chart)
+	}
+
+	if err = scaffold.MergeHelmRoleForResource(r, absProjectPath, roleScaffold); err != nil {
+		return fmt.Errorf("failed to update the RBAC manifest for resource (%v, %v): %v",
```

Comment:
Nit: To differentiate error messages to make debugging easier
```suggestion
		return fmt.Errorf("failed to merge rules in the RBAC manifest for resource (%v, %v): %v",
```